go/teams/list.go

```diff
@@ -493,38 +477,62 @@ func ComputeInvitelinkDisplayName(mctx libkb.MetaContext, team *Team, invite key
 	return name, nil
 }
 
-func addKeybaseInviteToRes(ctx context.Context, memb keybase1.TeamMemberDetails,
-	membs []keybase1.TeamMemberDetails) []keybase1.TeamMemberDetails {
-	for idx, existing := range membs {
-		if memb.Uv.Uid.Equal(existing.Uv.Uid) && !existing.Status.IsActive() {
-			membs[idx] = memb
-			return membs
-		}
-	}
-	return append(membs, memb)
-}
-
-func AnnotateTeamUsedInviteLogPoints(points []keybase1.TeamUsedInviteLogPoint, namePkgs map[keybase1.UID]libkb.UsernamePackage) (ret []keybase1.AnnotatedTeamUsedInviteLogPoint) {
+func annotateTeamUsedInviteLogPoints(points []keybase1.TeamUsedInviteLogPoint, namePkgs map[keybase1.UID]libkb.UsernamePackage) (ret []keybase1.AnnotatedTeamUsedInviteLogPoint) {
 	for _, point := range points {
 		username := namePkgs[point.Uv.Uid].NormalizedUsername
 		ret = append(ret, keybase1.AnnotatedTeamUsedInviteLogPoint{TeamUsedInviteLogPoint: point, Username: username.String()})
 	}
 	return ret
 }
 
-type AnnotatedInviteMap map[keybase1.TeamInviteID]keybase1.AnnotatedTeamInvite
+// GetAnnotatedInvitesAndMembersForUI returns annotated invites and members lists from the sigchain,
+// except that Keybase-type invites are treated as team members. This is for the purpose of
+// collecting them together in the UI.
+func GetAnnotatedInvitesAndMembersForUI(mctx libkb.MetaContext, team *Team,
```

Comment:
large changes here. Note it's returning every invite now, even cancelled/completed/obsolete ones.